src/viam/app/app_client.py

```diff
@@ -350,6 +394,10 @@ def __init__(self, channel: Channel, metadata: Mapping[str, str], location_id: O
     _channel: Channel
     _organization_id: Optional[str] = None
 
+    # TODO(RSDK-5569): this method will no longer be reliable when we're not authing as an org.
+    # consider what assumptions we _can_ make (e.g., if `list_orgs` only returns a single org
+    # then we can probably still rely on it for org id?), and also update other APIs to ask for
+    # an org id (optionally?) instead of just assuming we'll get it from this method.
     async def _get_organization_id(self) -> str:
         return self._organization_id if self._organization_id else await self._request_organization_id()
```

Comment:
Probably a perverse case, but do we want to request a new org_id if the current org_id is an empty string (`""`)? If that is a valid org_id for some reason then we need to check for `not None` explicitly since empty strings are falsy in python.

```suggestion
        return self._organization_id if self._organization_id is not None else await self._request_organization_id()
```